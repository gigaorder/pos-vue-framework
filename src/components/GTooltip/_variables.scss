@import "../../style/colors";

$border-radius-root: 4px !default;
$transition: (
        'fast-out-slow-in': cubic-bezier(0.4, 0, 0.2, 1),
        'linear-out-slow-in': cubic-bezier(0, 0, 0.2, 1),
        'fast-out-linear-in': cubic-bezier(0.4, 0, 1, 1),
        'ease-in-out': cubic-bezier(0.4, 0, 0.6, 1),
        'fast-in-fast-out': cubic-bezier(0.25, 0.8, 0.25, 1),
        'swing': cubic-bezier(0.25, 0.8, 0.5, 1)
) !default;

$tooltip-text-color: map-get($shades, 'white') !default;
$tooltip-border-radius: $border-radius-root !default;
$tooltip-font-size: 14px !default;
