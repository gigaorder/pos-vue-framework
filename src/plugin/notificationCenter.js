import Vue from 'vue';
import GNotification from '../components/GNotification/GNotification';

export default {
  install(Vue, options) {
    const notificationCenter = new Vue({
      ...GNotification,
      propsData: options ? parseOptions(options) : {}
    })
    notificationCenter.$mount(document.body.appendChild(document.createElement('div')));

    Vue.prototype.$notificationCenter = notificationCenter;
  }
}

function parseOptions(options) {
  const propsData = {}

  if(options.position) {
    const position = options.position.split('-');
    for (let i = 0; i < position.length; i++) {
      propsData[position[i]] = true
    }
  }

  if (options.width) propsData['width'] = options.width

  return propsData;
}