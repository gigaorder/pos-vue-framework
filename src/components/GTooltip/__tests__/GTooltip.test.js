import {createElement} from '../../../tests/setup'
const Vue = require('vue/dist/vue.common.js')
import plugin from 'vue'
Vue.use(plugin)

// Import your components after all of above import
import GTooltip from '../GTooltip'

describe('Tooltip', () => {
  function prepare(template) {
    return new Vue({
      template: template,
      components: { GTooltip },
    }).$mount(createElement())
  }

  describe('1) Render - ', () => {
    describe('Content - ', () => {
      it('Speech bubble content will be rendered if absolute is not exists', () => {
        let dom = prepare('<div><g-tooltip speech-bubble><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();

        dom = prepare('<div><g-tooltip absolute speech-bubble><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })
    })
  })

  describe('2) Position - ', () => {
    describe('Absolute - ', () => {
      it('Should set tooltip style with left: 0px, top: 0px', () => {
        // without unit
        let dom = prepare('<div><g-tooltip absolute><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();

        // with unit
        dom = prepare('<div><g-tooltip absolute><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })

      it('Should set tooltip style with left: 100px, top: 0px', () => {
        // without unit
        let dom = prepare('<div><g-tooltip absolute absolute-x="100"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();

        // with unit
        dom = prepare('<div><g-tooltip absolute absolute-x="100px"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })

      it('Should set tooltip style with left: 0px, top: 100px', () => {
        // wo unit
        let dom = prepare('<div><g-tooltip absolute absolute-y="100"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
        // w unit
        dom = prepare('<div><g-tooltip absolute absolute-y="100px"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })

      it('Should set tooltip style with left: 100px, top: 100px', () => {
        // wo unit
        let dom = prepare('<div><g-tooltip absolute absolute-x="100" absolute-y="100"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();

        // w unit
        dom = prepare('<div><g-tooltip absolute absolute-x="100px" absolute-y="100px"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })

      it('Should set tooltip style with left: 0px, top: 0px if absolute is not set', () => {
        const dom = prepare('<div><g-tooltip absolute-x="100" absolute-y="100"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })

      it('Should ignore all alignment setting if absolute is set', () => {
        let dom = prepare('<div style="padding-top: 500px; padding-left: 500px"><g-tooltip absolute top><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();

        dom = prepare('<div style="padding-top: 500px; padding-left: 500px"><g-tooltip absolute left><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();

        dom = prepare('<div style="padding-top: 500px; padding-left: 500px"><g-tooltip absolute right><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();

        dom = prepare('<div style="padding-top: 500px; padding-left: 500px"><g-tooltip absolute bottom><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })

      it('Should ignore all nudge setting if absolute is set', () => {
        const dom = prepare('<div style="padding-top: 500px; padding-left: 500px"><g-tooltip absolute nudge-top="200" nudge-right="200" nudge-bottom="200" nudge-left="200"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
        expect(dom.$el.outerHTML).toMatchSnapshot();
      })
    })

    describe('CalculatedLeft', () => {
      // TODO: calculatedLeftImplement
      // - Overflow
      // - Alignment Left, Right, NudgeLeft, NudgeRight

      // Pseudo
      // const props = {}
      // const dimensions = {}
      // const calculateXOverflow = {} // import fron menuable
      // expect(calculateLeftImplement(props, dimensions, calculateXOverflow)).toBe(100)
    })

    describe('CalculatedTop', () => {
      // TODO: calculatedTopImplement
      // - Overflow
      // - Alignment Top, Bottom, NudgeTop, NudgeBottom
    })
  })

  describe('3) Color - ', () => {
    const color = '#fafafa'
    it('Should be set for normal tooltip', () => {
      const dom = prepare(`<div><g-tooltip color="${color}"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should be set - speech bubble tooltip with implicit position is top by default', () => {
      const dom = prepare(`<div><g-tooltip color="${color}" speech-bubble><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should be set - speech bubble tooltip with explicit position is top by default', () => {
      const dom = prepare(`<div><g-tooltip color="${color}" speech-bubble top><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should be set - speech bubble tooltip with explicit position is left by default', () => {
      const dom = prepare(`<div><g-tooltip color="${color}" speech-bubble left><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should be set - speech bubble tooltip with explicit position is right by default', () => {
      const dom = prepare(`<div><g-tooltip color="${color}" speech-bubble right><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should be set - speech bubble tooltip with explicit position is bottom by default', () => {
      const dom = prepare(`<div><g-tooltip color="${color}" speech-bubble bottom><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })
  })

  describe('4) Behavior - ', () => {
    it('Should not toggle by click if open-on-hover is set', () => {
      const dom = prepare('<div><g-tooltip open-on-hover><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
      // if open-on-hover is not set, click event won't change display state of tooltip content
      dom.$el.querySelector('button').click()
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should be open by click if open-on-hover is not set while tooltip is close', (done) => {
      const dom = prepare('<div><g-tooltip><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
      // if open-on-hover is not set, click event won't change display state of tooltip content
      dom.$el.querySelector('button').click()
      setTimeout(() => {
        expect(dom.$el.outerHTML).toMatchSnapshot();
        done()
      }, 100);
    })

    it('Should be close by click if open-on-hover is not set while tooltip is open', (done) => {
      const dom = prepare('<div><g-tooltip><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
      // if open-on-hover is not set, click event won't change display state of tooltip content
      dom.$el.querySelector('button').click() // open
      dom.$el.querySelector('button').click() // close
      setTimeout(() => {
        expect(dom.$el.outerHTML).toMatchSnapshot();
        done()
      }, 100);
    })

    // Testing asynchronous code: https://jestjs.io/docs/en/asynchronous
    it('open-delay', (done) => {
      const delayTime = 1000 // 1s
      const dom = prepare(`<div><g-tooltip open-delay="${delayTime}"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      dom.$el.querySelector('button').click()
      expect(dom.$el.outerHTML).toMatchSnapshot();

      setTimeout(() => {
        expect(dom.$el.outerHTML).toMatchSnapshot();
        done()
      }, delayTime)
    })

    // Testing asynchronous code: https://jestjs.io/docs/en/asynchronous
    it('close-delay', (done) => {
      const delayTime = 1000 // 1s
      const dom = prepare(`<div><g-tooltip close-delay="${delayTime}"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>`);
      dom.$el.querySelector('button').click()
      expect(dom.$el.outerHTML).toMatchSnapshot();

      setTimeout(() => {
        expect(dom.$el.outerHTML).toMatchSnapshot();
        done()
      }, delayTime)
    })
  })

  describe('5) Max, min - ', () => {
    // max
    it('Should generate max-width: auto if max-width not set', () => {
      const dom = prepare('<div><g-tooltip><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should generate max-width: 100 if max-width=\"100\"', () => {
      const dom = prepare('<div><g-tooltip max-width="100"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })


    it('Should not generate min-width: (any) if min-width not set', () => {
      const dom = prepare('<div><g-tooltip><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })

    it('Should generate min-width: 100 if min-width=\"100\"', () => {
      const dom = prepare('<div><g-tooltip min-width="100"><template #activator="{on}"><button v-on="on">Click me</button></template><span>Tooltip content</span></g-tooltip></div>');
      expect(dom.$el.outerHTML).toMatchSnapshot();
    })
  })
})


