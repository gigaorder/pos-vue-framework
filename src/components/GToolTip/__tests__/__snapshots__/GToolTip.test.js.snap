// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip 1) Render -  Activator - Activator will be rendered if activator slot is set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 1) Render -  Activator - Activator won't be rendered if activator slot is not set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
    </div>
  </span>
</div>
`;

exports[`Tooltip 1) Render -  Content -  Content will be rendered if default slot is set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 1) Render -  Content -  Content won't be rendered if default slot is not set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 1) Render -  Content -  Speech bubble content will be rendered if absolute is not exists 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <div class="g-tooltip__content__speech-bubble g-tooltip__content__speech-bubble--bottom"
           style="border-top-color: #616161;"
      >
        <span>
          Tooltip content
        </span>
      </div>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 1) Render -  Content -  Speech bubble content will be rendered if absolute is not exists 2`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should ignore all alignment setting if absolute is set 1`] = `
<div style="padding-top: 500px; padding-left: 500px;">
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should ignore all alignment setting if absolute is set 2`] = `
<div style="padding-top: 500px; padding-left: 500px;">
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should ignore all alignment setting if absolute is set 3`] = `
<div style="padding-top: 500px; padding-left: 500px;">
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should ignore all alignment setting if absolute is set 4`] = `
<div style="padding-top: 500px; padding-left: 500px;">
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should ignore all nudge setting if absolute is set 1`] = `
<div style="padding-top: 500px; padding-left: 500px;">
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 0px, top: 0px 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 0px, top: 0px 2`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 0px, top: 0px if absolute is not set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 0px, top: 100px 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 100px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 0px, top: 100px 2`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 100px, top: 0px 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 100px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 100px, top: 0px 2`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 100px, top: 100px 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 100px; max-width: auto; opacity: 0; top: 100px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 2) Position -  Absolute -  Should set tooltip style with left: 100px, top: 100px 2`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="max-width: auto; opacity: 0; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 3) Color -  Should be set - speech bubble tooltip with explicit position is bottom by default 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 10px; z-index: 999; background-color: rgb(250, 250, 250); border-color: #fafafa; display: none;"
    >
      <div class="g-tooltip__content__speech-bubble g-tooltip__content__speech-bubble--top"
           style="border-bottom-color: #fafafa;"
      >
        <span>
          Tooltip content
        </span>
      </div>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 3) Color -  Should be set - speech bubble tooltip with explicit position is left by default 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(250, 250, 250); border-color: #fafafa; display: none;"
    >
      <div class="g-tooltip__content__speech-bubble g-tooltip__content__speech-bubble--right"
           style="border-left-color: #fafafa;"
      >
        <span>
          Tooltip content
        </span>
      </div>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 3) Color -  Should be set - speech bubble tooltip with explicit position is right by default 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(250, 250, 250); border-color: #fafafa; display: none;"
    >
      <div class="g-tooltip__content__speech-bubble g-tooltip__content__speech-bubble--left"
           style="border-right-color: #fafafa;"
      >
        <span>
          Tooltip content
        </span>
      </div>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 3) Color -  Should be set - speech bubble tooltip with explicit position is top by default 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(250, 250, 250); border-color: #fafafa; display: none;"
    >
      <div class="g-tooltip__content__speech-bubble g-tooltip__content__speech-bubble--bottom"
           style="border-top-color: #fafafa;"
      >
        <span>
          Tooltip content
        </span>
      </div>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 3) Color -  Should be set - speech bubble tooltip with implicit position is top by default 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(250, 250, 250); border-color: #fafafa; display: none;"
    >
      <div class="g-tooltip__content__speech-bubble g-tooltip__content__speech-bubble--bottom"
           style="border-top-color: #fafafa;"
      >
        <span>
          Tooltip content
        </span>
      </div>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 3) Color -  Should be set for normal tooltip 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(250, 250, 250); border-color: #fafafa; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 4) Behavior -  Should be close by click if open-on-hover is not set while tooltip is open 1`] = `
<div>
  <div class="g-tooltip__content"
       style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
  >
    <span>
      Tooltip content
    </span>
  </div>
  <div class="g-tooltip__activator">
    <button>
      Click me
    </button>
  </div>
  <span class="g-tooltip">
  </span>
</div>
`;

exports[`Tooltip 4) Behavior -  Should be open by click if open-on-hover is not set while tooltip is close 1`] = `
<div>
  <div class="g-tooltip__content"
       style="left: 0px; max-width: auto; opacity: 1; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161;"
  >
    <span>
      Tooltip content
    </span>
  </div>
  <div class="g-tooltip__activator">
    <button>
      Click me
    </button>
  </div>
  <span class="g-tooltip">
  </span>
</div>
`;

exports[`Tooltip 4) Behavior -  Should not toggle by click if open-on-hover is set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 4) Behavior -  close-delay 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 4) Behavior -  close-delay 2`] = `
<div>
  <div class="g-tooltip__content"
       style="left: 0px; max-width: auto; opacity: 1; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161;"
  >
    <span>
      Tooltip content
    </span>
  </div>
  <div class="g-tooltip__activator">
    <button>
      Click me
    </button>
  </div>
  <span class="g-tooltip">
  </span>
</div>
`;

exports[`Tooltip 4) Behavior -  open-delay 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 4) Behavior -  open-delay 2`] = `
<div>
  <div class="g-tooltip__content"
       style="left: 0px; max-width: auto; opacity: 1; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161;"
  >
    <span>
      Tooltip content
    </span>
  </div>
  <div class="g-tooltip__activator">
    <button>
      Click me
    </button>
  </div>
  <span class="g-tooltip">
  </span>
</div>
`;

exports[`Tooltip 5) Max, min -  Should generate max-width: 100 if max-width="100" 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: 100px; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 5) Max, min -  Should generate max-width: auto if max-width not set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 5) Max, min -  Should generate min-width: 100 if min-width="100" 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; min-width: 100px; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;

exports[`Tooltip 5) Max, min -  Should not generate min-width: (any) if min-width not set 1`] = `
<div>
  <span class="g-tooltip">
    <div class="g-tooltip__content"
         style="left: 0px; max-width: auto; opacity: 0; top: 0px; z-index: 999; background-color: rgb(97, 97, 97); border-color: #616161; display: none;"
    >
      <span>
        Tooltip content
      </span>
    </div>
    <div class="g-tooltip__activator">
      <button>
        Click me
      </button>
    </div>
  </span>
</div>
`;
